distributed/scheduler.py

```diff
@@ -8561,7 +8592,7 @@ def _materialize_graph(
 
     # - Add in deps for any tasks that depend on futures
     for k, futures in fut_deps.items():
-        dependencies[k].update(f.key for f in futures)
+        dependencies[k].update(f.key for f in futures if f.key != k)
```

Comment:
Would be nice to link to the unit test that demonstrates the need for this edge case test